rohmu/object_storage/azure.py

```diff
@@ -156,36 +162,41 @@ def copy_file(self, *, source_key, destination_key, metadata=None, **kwargs):
                     )
                 )
 
-    def get_metadata_for_key(self, key):
+    def get_metadata_for_key(self, key: str) -> Metadata:
         path = self.format_key_for_backend(key, remove_slash_prefix=True, trailing_slash=False)
         items = list(self._iter_key(path=path, with_metadata=True, deep=False))
         if not items:
             raise FileNotFoundFromStorageError(path)
         expected_name = path.rsplit("/", 1)[-1]
+        item: Optional[IterKeyItem]
         for item in items:
+            assert item is not None  # for mypy
             # We expect single result but Azure listing is prefix match so we need to explicitly
             # look up the matching result
+            item_name: str
             if item.type == KEY_TYPE_OBJECT:
-                item_name = item.value["name"]
+                item_name = cast(Dict[str, str], item.value)["name"]
             else:
-                item_name = item.value
+                item_name = cast(str, item.value)
             if item_name.rstrip("/").rsplit("/", 1)[-1] == expected_name:
                 break
         else:
             item = None
         if not item or item.type != KEY_TYPE_OBJECT:
             raise FileNotFoundFromStorageError(path)  # not found or prefix
-        return item.value["metadata"]
+        return cast(Dict[str, Any], item.value)["metadata"]
```

Comment:
nit: Is it possible to cast to `Metadata` instead of `Dict[str, Any]`?

Reply:
Yes, but here the cast is on the "wrapper" object from which we extract the metadata.

Technically `Metadata` is just `Dict[str, Any]` but since I'm using it to refer to whatever is found under the `metadata` key I want to avoid using it outside that context. In the future the `Metadata` might become some `TypedDict` and suddenly it would not work correctly as used here.